internal/util/state/state.go

```diff
@@ -32,7 +32,7 @@ type State struct {
 	// never persisted
 	TelemetryLocked   bool      `json:"-"`
 	Start             time.Time `json:"-"`
-	LatestVersion     string    `json:"-"` // empty if IsUpdateAvailable is false
+	LatestVersion     string    `json:"-"` // empty if telemetry is disabled; latest version reported by telemetry otherwise
```

Comment:
That first part should not be true anymore. We should not set `LatestVersion` to the empty string if telemetry is disabled because there is no need to – we don't check it's value if it is empty or not without checking `UpdateAvailable` first

Reply:
fixed